### PR TITLE
GNA shorten affine filter

### PR DESCRIPTION
**Pull request by esmirno**
memory and compute time improvements for concat align filter used in gna-plugin

### CODE REVIEW

File: inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp [hunk 1]
```diff
@@ -17,6 +17,10 @@ void LayerTestsCommon::Run() {
     LoadNetwork();
     Infer();
     Validate();
+
+    if (targetDevice == std::string{CommonTestUtils::DEVICE_GPU}) {
```
Comment (iefode) [inline inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp:21]: These changes is in destructor. Please remove

File: inference-engine/tests_deprecated/functional/ie_tests/src/custom_matcher.cpp [hunk 2]
```diff
@@ -368,7 +368,7 @@ void Regression::Matchers::CustomMatcher::checkResult() {
                 maxAverageDiff = std::max(maxAverageDiff, avgDiff / avgFrames);
 
                 //TODO: add test_log parse from command line
-// #define TEST_LOG
+#define TEST_LOG
```
Comment (iefode) [inline inference-engine/tests_deprecated/functional/ie_tests/src/custom_matcher.cpp:371]: You forgot to add `//`

File: inference-engine/tests_deprecated/unit/engines/gna/gna_matcher.cpp [hunk 3]
```diff
@@ -66,6 +66,33 @@ class NullAllocator : public IAllocator {
     }
 };
 
+template<class T>
```
Comment (iefode) [inline inference-engine/tests_deprecated/unit/engines/gna/gna_matcher.cpp:69]: Can you move changed tests to the new test infra? We should migrate tests to the new infra?

File: inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp (hunk 1 above)
Comment (esmirno) [reply inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp:21]: in destructor it is not correct place

File: inference-engine/tests_deprecated/functional/ie_tests/src/custom_matcher.cpp (hunk 2 above)
Comment (esmirno) [reply inference-engine/tests_deprecated/functional/ie_tests/src/custom_matcher.cpp:371]: lets wait for error from cpplint then - if not wouldnt fix it now

File: inference-engine/tests_deprecated/unit/engines/gna/gna_matcher.cpp (hunk 3 above)
Comment (esmirno) [reply inference-engine/tests_deprecated/unit/engines/gna/gna_matcher.cpp:69]: for unit tests we dont need your infrastructure - at all.

File: inference-engine/tests/functional/plugin/shared/include/subgraph_tests/func_concat_2_inputs.hpp [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+class NonTrivialConcat2Inputs : public testing::WithParamInterface<NonTrivialConcatParams>,
+                        public LayerTestsUtils::LayerTestsCommon {
+ public:
+    static std::string getTestCaseName(testing::TestParamInfo<NonTrivialConcatParams> obj);
+    float threshold(const InferenceEngine::Precision & precision) override {
+        return fp32Threshold;
+    }
+
+    InferenceEngine::Blob::Ptr GenerateInput(const InferenceEngine::InputInfo &info) const override {
```
Comment (esmirno) [inline inference-engine/tests/functional/plugin/shared/include/subgraph_tests/func_concat_2_inputs.hpp:27]: @bsochack please use that override for power

**Comment by ilya-lavrenov** [comment]
@dorloff @nkogteva  should we close this?

**Pull request closed without merging**